envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/JwtFilterFactory.kt

```diff
@@ -48,7 +48,9 @@ class JwtFilterFactory(
     }
 
     private fun shouldCreateFilter(group: Group): Boolean {
-        return group.proxySettings.incoming.endpoints.any { it.oauth != null } && properties.providers.isNotEmpty()
+        return group.proxySettings.incoming.endpoints.any {
```

Comment:
Can we change the order so if no providers configured we won't iterate: ```
properties.providers.isNotEmpty() && group.proxySettings.incoming.endpoints.any {
            it.oauth != null || it.clients.any { client -> client.selector in selectorToOAuthProvider.keys }
        } 
```
also you could extract `it.clients.any { client -> client.selector in selectorToOAuthProvider.keys }` this to separate method